relayer/app/state.go

```diff
@@ -0,0 +1,110 @@
+package relayer
+
+import (
+	"encoding/json"
+	"os"
+	"sync"
+
+	"github.com/omni-network/omni/lib/errors"
+
+	"github.com/cometbft/cometbft/libs/tempfile"
+)
+
+// State represents the state of the relayer. It keeps track of the last submitted height for each source chain on each destination chain.
```

Comment:
not "last submitted height" since we don't submit empty block. Rather "last successfully process block (whether submitted or skipped due to being empty).